pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/predicate/NotEqualsPredicateEvaluatorFactory.java

```diff
@@ -75,8 +75,12 @@ public static BaseRawValueBasedPredicateEvaluator newRawValueBasedEvaluator(NEqP
       _nonMatchingDictId = dictionary.indexOf(nEqPredicate.getNotEqualsValue());
       if (_nonMatchingDictId >= 0) {
         _nonMatchingDictIds = new int[]{_nonMatchingDictId};
+        if (dictionary.length() == 1) {
```

Comment:
Shouldn't this be _nonMatchingIds == dictionary.length()?

Reply:
You mean **_nonMatchingDictIds.length == dictionary.length()**? **_nonMatchingDictIds.length** is always 1 here.